### ISSUE TITLE
chance and faker in devDependencies?

### ISSUE THREAD

**Issue by tauren** [json-schema-faker/json-schema-faker#99]
Shouldn't `chance` and `faker` be regular dependencies, not a devDependency?

I run `npm install` in my consuming app (which is dependent on `json-schema-faker@0.2.9`):

```
:npm install
json-schema-faker@0.2.9 node_modules/json-schema-faker
├── deref@0.6.3 (deep-extend@0.4.1)
└── randexp@0.4.2 (discontinuous-range@1.0.0, ret@0.1.11)
Thu Feb 04 12:38: 16 PST APRYFD59D9B480 ~/stash/nidr/playbook-rulebook  
```

I then use `babeljs` to transpile my consuming app to ES5:

```
:gulp build
[12:38:26] Requiring external module babel-core/register
[12:38:27] Using gulpfile ~/stash/nidr/playbook-rulebook/gulpfile.babel.js
[12:38:27] Starting 'transpile'...
[12:38:27] Starting 'bundle'...
[12:38:27] Starting 'schemas'...
[12:38:28] Finished 'transpile' after 1.26 s
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: Cannot find module 'chance' from '/Users/tmil11/stash/nidr/playbook-rulebook/node_modules/json-schema-faker'
    at /Users/tmil11/stash/nidr/playbook-rulebook/node_modules/browserify/node_modules/resolve/lib/async.js:46:17
    at process (/Users/tmil11/stash/nidr/playbook-rulebook/node_modules/browserify/node_modules/resolve/lib/async.js:173:43)
    at ondir (/Users/tmil11/stash/nidr/playbook-rulebook/node_modules/browserify/node_modules/resolve/lib/async.js:188:17)
    at load (/Users/tmil11/stash/nidr/playbook-rulebook/node_modules/browserify/node_modules/resolve/lib/async.js:69:43)
    at onex (/Users/tmil11/stash/nidr/playbook-rulebook/node_modules/browserify/node_modules/resolve/lib/async.js:92:31)
    at /Users/tmil11/stash/nidr/playbook-rulebook/node_modules/browserify/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:95:15)
```

**Comment by pateketrueke** [comment]
Yes and no.

First sorry for the missing docs about that (I'll update the README asap), but this was an issue: https://github.com/json-schema-faker/json-schema-faker/issues/69

Basically jsf SHOULD work without faker/chance until you use the faker/chance keywords, that's it.

Before 0.2.9 some basic generators was tied to faker's helpers.

This is a breaking change towards 0.3.0, sorry for that.

If you think this was a big mistake (and I think it was without announcing it) please let me know.

**Comment by owencraig** [comment]
I think the bigger issue for this is that moving it to be optional is a breaking change and was slated as a patch. So anyone using a pathspec like ~0.2.0 or even ^0.0.1 expecting only bug fixes or feature is getting a breaking change. 

I wonder if potentially making faker and chance a peerDependency would be more explicit and put it on a path to a full version bump

**Comment by pateketrueke** [comment]
Totally agree, it was my mistake, I'll deprecate that version asap and release 0.2.10 right now!

**Comment by heckj** [comment]
FWIW: this broke our downstream builds as well... (https://travis-ci.org/RackHD/on-http/builds/107051850). Appreciate you pushing out an update ASAP!

**Comment by pateketrueke** [comment]
Released 0.2.10, sorry for that!
